File: algorithms/643. Maximum Average Subarray I/solution.cc
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double maxsum = 0, ksum = 0;
        for ( int i = 0; i < k; ++i )
            maxsum = ksum = ksum + nums[i];
        for ( int i = k; i < nums.size(); ++i ) {
            ksum += nums[i] - nums[i-k];
            maxsum = max(maxsum, ksum);
        }
        return maxsum / k;
    }
};

File: algorithms/643. Maximum Average Subarray I/solution.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        maxsum = ksum = sum(itertools.islice(nums, 0, k))
        for i in xrange(k, len(nums)):
            ksum += nums[i] - nums[i-k]
            maxsum = max(maxsum, ksum)
        return maxsum / float(k)
